$OpenBSD: patch-spooler_Spooler_cpp,v 1.1 2011/03/29 18:00:54 ajacoutot Exp $

XXX to be revisited
See also patch-installer_linux_spooler_cfg

--- spooler/Spooler.cpp.orig	Tue Mar 29 00:04:09 2011
+++ spooler/Spooler.cpp	Tue Mar 29 19:07:28 2011
@@ -1070,6 +1070,7 @@ int main(int argc, char *argv[]) {
 	if (g_lpConfig->HasWarnings())
 		LogConfigErrors(g_lpConfig, g_lpLogger);
 
+#ifdef __linux__
 	// detect linuxthreads, which is too broken to correctly run the spooler
 	if (!bForked) {
 		char buffer[256];
@@ -1081,6 +1082,13 @@ int main(int argc, char *argv[]) {
 			g_lpLogger->Log(EC_LOGLEVEL_FATAL, "WARNING: the zarafa-spooler will only be able to send one message at a time.");
 		}
 	}
+#endif
+
+// let's be safe because our userland pthreads(3) sucks
+#ifdef __OpenBSD__
+	g_lpConfig->AddSetting("max_threads","1");
+#endif
+
 
 	// set socket filename
 	if (!szPath)
